### PR TITLE
Include SVP transactions as possible transaction types to be used by the Bridge

### PR DESCRIPTION
**Pull request by marcos-iov**
## Description
- Create a FederationContext class, to hold information about active and retiring federations
- Refactor PegUtils methods to receive a FederationContext object instead of each element separately
- Update getTransactionType method to detect SVP related transactions
- Update PegTxType enum to add SVP types
- Update registerBtcTransaction to detect and process SVP related transactions

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)

* **Other information**:

### CODE REVIEW

**Review by apancorb — COMMENTED** [review]
Looking good to me

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationContextTest.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    void getRetiringFederation() {
+        assertTrue(federationContext.getRetiringFederation().isEmpty());
+    }
+
+    @Test
+    void setRetiringFederation() {
+        federationContext.setRetiringFederation(retiringFederation);
+
+        assertTrue(federationContext.getRetiringFederation().isPresent());
+        assertEquals(retiringFederation, federationContext.getRetiringFederation().get());
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationContextTest.java:41]: nit:
```
Federation retiringFederation = federationContext.getRetiringFederation();
assertTrue(retiringFederation.isPresent());
assertEquals(retiringFederation, retiringFederation.get());
```

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationContextTest.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        assertTrue(federationContext.getLastRetiredFederationP2SHScript().isEmpty());
+    }
+
+    @Test
+    void setLastRetiredFederationP2SHScript() {
+        Script lastRetiredFederationP2SHScript = retiringFederation.getP2SHScript();
+        federationContext.setLastRetiredFederationP2SHScript(lastRetiredFederationP2SHScript);
+
+        assertTrue(federationContext.getLastRetiredFederationP2SHScript().isPresent());
+        assertEquals(lastRetiredFederationP2SHScript, federationContext.getLastRetiredFederationP2SHScript().get());
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationContextTest.java:55]: same here:

Script lastRetiredFederationP2SHScript = federationContext.getLastRetiredFederationP2SHScript();
assertTrue(lastRetiredFederationP2SHScript.isPresent());
assertEquals(lastRetiredFederationP2SHScript, lastRetiredFederationP2SHScript.get());

File: rskj-core/src/main/java/co/rsk/peg/federation/FederationContext.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import java.util.stream.Stream;
+
+public class FederationContext {
+    private final Federation activeFederation;
+    private Federation retiringFederation;
+    private Script lastRetiredFederationP2SHScript;
+
+    public FederationContext(Federation activeFederation) {
+        this.activeFederation = activeFederation;
+    }
```
Comment (nathanieliov) [inline rskj-core/src/main/java/co/rsk/peg/federation/FederationContext.java:16]:
```suggestion
    private FederationContext(Federation activeFederation) {
        this.activeFederation = activeFederation;
    }

    public static FederationContext of(Federation activeFederation) {
        return new FederationContext(activeFederation);
    }
```

File: rskj-core/src/main/java/co/rsk/peg/federation/FederationSupport.java [hunk 4]
```diff
@@ -16,34 +16,154 @@
 public interface FederationSupport {
 
     Federation getActiveFederation();
```
Comment (nathanieliov) [inline rskj-core/src/main/java/co/rsk/peg/federation/FederationSupport.java:18]:
```suggestion
/**
     * Returns the currently active federation.
     * See getActiveFederationReference() for details.
     * @return the active federation.
     */
    Federation getActiveFederation();
    Federation getActiveFederation();
```

**Pull request closed without merging**